eth-custodian/contracts/EthCustodianProxy.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+import '@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import 'rainbow-bridge-sol/nearprover/contracts/INearProver.sol';
+import 'rainbow-bridge-sol/nearprover/contracts/ProofDecoder.sol';
+import 'rainbow-bridge-sol/nearbridge/contracts/Borsh.sol';
+import {EthCustodian} from './EthCustodian.sol';
+import {SelectivePausableUpgradable} from './SelectivePausableUpgradable.sol';
+
+contract EthCustodianProxy is
+    Initializable,
+    UUPSUpgradeable,
+    AccessControlUpgradeable,
+    SelectivePausableUpgradable
+{
+    using SafeERC20 for IERC20;
+    using Borsh for Borsh.Data;
+    using ProofDecoder for Borsh.Data;
+
+    bytes32 public constant PAUSABLE_ADMIN_ROLE = keccak256('PAUSABLE_ADMIN_ROLE');
+    bytes32 public constant UNPAUSABLE_ADMIN_ROLE = keccak256('UNPAUSABLE_ADMIN_ROLE');
+
+    uint constant UNPAUSED_ALL = 0;
+    uint constant PAUSED_DEPOSIT_TO_EVM = 1 << 0;
+    uint constant PAUSED_DEPOSIT_TO_NEAR = 1 << 1;
+    uint constant PAUSED_WITHDRAW = 1 << 2;
+    uint constant PAUSED_WITHDRAW_PRE_MIGRATION = 1 << 3;
+
+    event Withdrawn(address indexed recipient, uint128 amount);
+
+    error AlreadyMigrated();
+    error ProducerAccountIdTooLong(bytes newProducerAccount);
+    error ProofFromPostMergeBlock();
+
+    struct BurnResult {
```

Comment:
Where this struct is used?

Reply:
Leftover from another approach, removed f17bc73e4a13a4e86aa61c0691f4ba82231f3e6d